lib/rspec/matchers/built_in/output.rb

```diff
@@ -223,21 +225,47 @@ def capture(block)
           # thread, fileutils, etc), so it's worth delaying it until this point.
           require 'tempfile'
 
+          # This is.. awkward-looking. But it's written this way because of how
+          # compound matchers work - we essentially need to be able to tell if
+          # we're in an _inner_ matcher, so we can pass the stream-output along
+          # to the outer matcher for further evaluation in that case. Added to
+          # that, it's fairly difficult to _tell_, because the only actual state
+          # we have access to is the stream itself, and in the case of stderr,
+          # that stream is really a RSpec::Support::StdErrSplitter (which is why
+          # we're testing `is_a?(File)` in such an obnoxious way).
+          inner_matcher = stream.to_io.is_a?(File)
```

Comment:
Is there not a way to assert against one of our internal classes here? checking to see if we're already in a splitter etc

Reply:
Only in some of the cases. We're interacting with whatever's in `$stderr` and `$stdout`, so I think we're reliably getting an internal class for the former but not the latter. Wrapping with an internal class for the latter as well did occur to me as a potential solution, but it was large and involved enough that I wasn't willing to attempt it for such a minor issue :-\